js/mviewer.js

```diff
@@ -2640,12 +2733,26 @@ mviewer = (function () {
             var tabs = tab.parent().find("li");
             var info = $(tab.find("a").attr("href"));
 
+            _sourceSelectOverlay.getFeatures().forEach(feature => {
+                if (feature.get("mviewerid") === layerid) {
+                    _sourceSelectOverlay.removeFeature(feature);
+                    _sourceSubSelectOverlay.getFeatures().forEach(subFeature => {
+                        if (feature.ol_uid == subFeature.ol_uid) {
+                            _sourceSubSelectOverlay.removeFeature(subFeature)
+                        }
+                    })
+                }
+            })
+
             if ( tabs.length === 1 ) {
                 tab.remove();
                 info.remove();
                 if (panel.hasClass("active")) {
                     panel.toggleClass("active");
                 }
+                // only necessary for fallback pin (GetFeatureInfo without geometry).
+                // could be improved by checking if one of the displayed layers is concerned
```

Comment:
need issue instead of comment here if you advise to improve something.

Reply:
Got rid of the comment and improved the behavior ;-)